docs/source/daml/reference/contract-keys.rst

```diff
@@ -71,7 +82,18 @@ lookupByKey
 
 Use ``lookupByKey`` to check whether a contract with the specified key exists. If it does exist, ``lookupByKey`` returns the ``Some contractId``, where ``contractId`` is the ID of the contract; otherwise, it returns ``None``.
 
-``lookupByKey`` needs to be authorized by **all** maintainers of the contract you are trying to lookup. The reason for that is denial of service protection. Without this restriction, a malicious Party could legitimately request another Party to validate any number of negative contract key lookups without that other Party ever having signed anything.
+``lookupByKey`` is conceptually equivalent to
+
+.. code-block:: daml
+
+   lookupByKey k = do
+     canSee <- canSeeContractWithKey k
+     if canSee then
+       Some <$> fetchByKey k
```

Comment:
Not quite. `fetchByKey` returns a tuple `(cid, args)` so it would be `(Some . fst) <$> fetchByKey k`

Reply:
Replaced with code I actually tested this time.